lib/RefBackend/RefBackend.cpp

```diff
@@ -163,9 +168,11 @@ static LogicalResult mungeFunction(
     auto supportedFuncsEnd = supportedConsumeFuncReturnFuncs.end();
     std::string funcName = getConsumeReturnFunctionNameForReturnTypes(retTypes);
     if (supportedConsumeFuncReturnFuncs.find(funcName) == supportedFuncsEnd) {
-      op.emitError("must have one return value of memref types or scalar types "
-                   "of i32, i64, f32, f64, i1, or two return values of memref "
-                   "f32 and i64, or three return values of memref f32");
+      op.emitError(
+          "must have one return value of memref types or scalar types "
+          "of i32, i64, f32, f64, i1, or two return values of memref "
+          "f32 and i64, or two return values of memref f32, or two "
+          "return values of memref f64, or three return values of memref f32");
```

Comment:
nit: I wonder if it would be more clear to simply list the actual types rather than describe them. Such as

> Supported return types: 
> mri1, mri32, mri64, mrf32, mrf64, i64, f32, f64, 
> (mrf32, mri64), (mrf32, mrf32), (mrf64, mrf64),
> (mrf32, mrf32, mrf32)

This will make it a bit easier to keep this comment up to date and accurate. At the moment it gives the wrong information in the first sentence because values of `i32` and `i1` are not supported, only memrefs.